app/src/components/blocks/deleteDocumentsStage/DeleteDocumentsStage.tsx

```diff
@@ -83,8 +107,17 @@ function DeleteDocumentsStage({
                     passNavigate(routes.HOME);
                 }
             }
-        } else {
-            setStage(LG_RECORD_STAGE.RECORD);
+        } else if (fieldValues.deleteDocs === 'no') {
+            if (userType === USER_ROLE.GP) {
+                if (setStage) {
+                    setStage(LG_RECORD_STAGE.RECORD);
+                }
+            }
+            if (userType === USER_ROLE.PCSE) {
+                if (setIsDeletingDocuments) {
+                    setIsDeletingDocuments(false);
+                }
+            }
```

Comment:
There are quite alot of if else statements nested, including the big one for 'no' and 'yes' options, these can be seperated into more functions such as handleYesOption() handleNoOption()

Also maybe these no and yes option values can be enums? It's always better for type safety